packages/compiler/core/parser.ts

```diff
@@ -515,7 +515,13 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
     const templateParameters = parseTemplateParameterList();
 
     let mixes: TypeReferenceNode[] = [];
-    if (token() === Token.Identifier) {
+    if (token() === Token.ExtendsKeyword) {
+      // error condition
+      error({ code: "token-expected", messageId: "mixesNotExtends" });
+      parseErrorInNextFinishedNode = false;
```

Comment:
Hmm, do we end up with no node with an error flag here?

Reply:
I believe a test with this case would hit the invariant check on this if so. I think we would need to restore parseErrorInNextFinishedNode to true after parseList to make the interface node have an error since none of its children will for this error.

Reply:
Or maybe instead of that, do the error() after the parseList.

Reply:
I have a fix for this and will send a small PR that fixes this and a couple of other minor things I spotted nearby.